src/sync/sync_session.cpp

```diff
@@ -530,6 +534,9 @@ void SyncSession::create_sync_session()
             throw sync::BadServerUrl();
         }
         session_config.service_identifier += "/realm-sync";
+        // FIXME: Java needs the fully resolved URL for proxy support, but we also need it before
```

Comment:
Why? Only the hostname and port matter for `ProxySelector`, and those are already known on `App`.

Reply:
Ok. I didn't look closely into exactly what is required for the proxy selector to work. I just noticed that right now we are using the fully qualified URL.

I probably still think exposing the full URL would be nice for debugging, but isn't super critical.

Reply:
The URL for sync sessions is always going to be the same - the base URL of their `App` plus `/realm-sync`.